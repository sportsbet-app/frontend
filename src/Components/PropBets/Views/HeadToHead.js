//import hooks and react dependencies
import React, { useState } from 'react'
import Select from 'react-select'
//import styling and assets
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faUser, faChartLine, faSyncAlt } from '@fortawesome/free-solid-svg-icons'
import { statOptions, playerOptions, statTypeOption } from '../../../Assets/DummyData'
import { StyledButton, Descriptor, Flex, CountDisplay } from '../styledComponents'

//collecting props from PropForm for this functional component
const HeadToHead = props => {
  //setup state variables  
  const [count, setCount] = useState(0)
  const [statType, setStatType] = useState(statTypeOption[0])
  const [selectedOption, setSelectedOption] = useState(null)
  const [playerOneSelection, setPlayerOneSelection] = useState(null)
  const [playerTwoSelection, setPlayerTwoSelection] = useState(null)

  //function to reset state
  const clearOptions = _ => {
    setCount(0)
    setSelectedOption(null)
    setPlayerOneSelection(null)
    setPlayerTwoSelection(null)
  }
  //find the item with an id of wrap and toggle the class if there is no class on the htmlelement that is calling this function
  const displayBetSlip = () => {
    const container = document.getElementById('wrap')
    if (container.className.length !== 0) {
      container.classList.toggle('wrapper')
    }
  }
  //variables to take in icon images for styling
  const rotate = <FontAwesomeIcon icon={faSyncAlt} />
  const player = <FontAwesomeIcon icon={faUser} />
  const stat = <FontAwesomeIcon icon={faChartLine} />

  return (
    <>
      <Flex column>
        <Select
          className='ninety'
          placeholder={player}
          value={playerOneSelection}
          onChange={(sel) => setPlayerOneSelection(sel)}
          options={props.players}
          isSearchable={true}
        />

        <Flex >
          <i style={{ marginRight: '2rem' }} onClick={() => {
            setPlayerOneSelection(playerTwoSelection)
            setPlayerTwoSelection(playerOneSelection)

          }}>{rotate}</i>
          <Descriptor>Will have more than</Descriptor>
        </Flex>

        <Select
          className='ninety'
          placeholder={player}
          value={playerTwoSelection}
          onChange={(sel) => setPlayerTwoSelection(sel)}
          options={props.players}
          isSearchable={true}
        />
      </Flex>

      <Flex>
        <Select
          className='ninety'
          placeholder={stat}
          value={selectedOption}
          onChange={(sel) => setSelectedOption(sel)}
          options={statOptions}
          isSearchable={true}
        />
      </Flex>

      <CountDisplay>
        <p>{count}</p>
        <Select
          value={statType}
          onChange={(sel) => setStatType(sel)}
          options={statTypeOption}
          isSearchable={true}

        />
      </CountDisplay>

      <Flex spaceAJ>
        <StyledButton
            primary active={!selectedOption && !playerOneSelection && !playerTwoSelection}
            onClick={clearOptions}>Clear</StyledButton>
        <StyledButton
          primary active={!selectedOption || !playerOneSelection || !playerTwoSelection}
          disabled={!selectedOption || !playerOneSelection || !playerTwoSelection}
          onClick={() => {
            props.setBetSlip({
              count, selectedOption, playerOneSelection, playerTwoSelection,
              playerCount: 2
            })
            clearOptions()
            displayBetSlip()
          }}>Send To Betslip</StyledButton>
      </Flex>
    </>
  )
}

export default HeadToHead