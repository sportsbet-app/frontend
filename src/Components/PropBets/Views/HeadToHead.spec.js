import React from 'react'
//import dependencies for tests
import { render, fireEvent, act } from '@testing-library/react'

//import renderer for snapshot
import renderer from 'react-test-renderer'
//import component
import HeadToHead from './HeadToHead'

//create test data to use 
const dataPlayer = { 
  players: 'test',
  playerOneSelection: 'test number 1',
  playerTwoSelection: 'test number 2',
}

const dataBet = {

}

//test to check that the app will display when called.
describe('<HeadToHead />', () => {
  it('should display', () => {
    //render the component
    render(<HeadToHead players = { dataPlayer } setBetSlip = { dataBet } />)
    
  })
  //create a snapshot
  it('matches snapshot', () => {
    const tree = renderer.create(<HeadToHead players = { dataPlayer } setBetSlip = { dataBet }/>); // generates a DOM tree

    // snapshots are a JSON representation of the DOM tree
    expect(tree.toJSON()).toMatchSnapshot();
  });
  //check if the click works 
  //figure out whats being displayed as the onclick
  // it('clicks properly', () => {
  //   // setup render for the page
  //   const { getByText, findByAltText } = render(<HeadToHead players = { dataPlayer } setBetSlip = { dataBet } playerOneSelection = { dataPlayer.playerOneSelection }/>)
  //   //grab the item being clicked
  //   const click = getByText(/test/i)
  //   //call the function being clicked using act
  //   act( () => {
  //     fireEvent.click(click)
  //   })
  //   //create a variable for the find all to grab the first item
  //   const flip = findByAltText(/test/i)[0].toString()
  //   //check that variable is the player two selection
  //   expect(flip).toBe('test number 2')
  // })
})