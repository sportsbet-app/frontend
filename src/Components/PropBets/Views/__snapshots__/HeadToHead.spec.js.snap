// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeadToHead /> matches snapshot 1`] = `
Array [
  <div
    className="sc-ifAKCX jUueog"
  >
    <div
      className="ninety css-2b097c-container"
      onKeyDown={[Function]}
    >
      <div
        className=" css-yk16xz-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className=" css-1hwfws3"
        >
          <div
            className=" css-1wa3eu0-placeholder"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-user fa-w-14 "
              data-icon="user"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
          <div
            className="css-w8afj7-Input"
          >
            <div
              className=""
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                disabled={false}
                id="react-select-6-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "label": "input",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className=" css-1g48xl4-IndicatorsContainer"
        >
          <span
            className=" css-1okebmr-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            className=" css-tlfecz-indicatorContainer"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-tj5bde-Svg"
              focusable="false"
              height={20}
              viewBox="0 0 20 20"
              width={20}
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-ifAKCX bPsgee"
    >
      <i
        onClick={[Function]}
        style={
          Object {
            "marginRight": "2rem",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-sync-alt fa-w-16 "
          data-icon="sync-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M370.72 133.28C339.458 104.008 298.888 87.962 255.848 88c-77.458.068-144.328 53.178-162.791 126.85-1.344 5.363-6.122 9.15-11.651 9.15H24.103c-7.498 0-13.194-6.807-11.807-14.176C33.933 94.924 134.813 8 256 8c66.448 0 126.791 26.136 171.315 68.685L463.03 40.97C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.749zM32 296h134.059c21.382 0 32.09 25.851 16.971 40.971l-41.75 41.75c31.262 29.273 71.835 45.319 114.876 45.28 77.418-.07 144.315-53.144 162.787-126.849 1.344-5.363 6.122-9.15 11.651-9.15h57.304c7.498 0 13.194 6.807 11.807 14.176C478.067 417.076 377.187 504 256 504c-66.448 0-126.791-26.136-171.315-68.685L48.97 471.03C33.851 486.149 8 475.441 8 454.059V320c0-13.255 10.745-24 24-24z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </i>
      <h3
        className="sc-dnqmqq byWLMA"
      >
        Will have more than
      </h3>
    </div>
    <div
      className="ninety css-2b097c-container"
      onKeyDown={[Function]}
    >
      <div
        className=" css-yk16xz-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className=" css-1hwfws3"
        >
          <div
            className=" css-1wa3eu0-placeholder"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-user fa-w-14 "
              data-icon="user"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
          <div
            className="css-w8afj7-Input"
          >
            <div
              className=""
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                disabled={false}
                id="react-select-7-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "label": "input",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className=" css-1g48xl4-IndicatorsContainer"
        >
          <span
            className=" css-1okebmr-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            className=" css-tlfecz-indicatorContainer"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-tj5bde-Svg"
              focusable="false"
              height={20}
              viewBox="0 0 20 20"
              width={20}
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-ifAKCX bPsgee"
  >
    <div
      className="ninety css-2b097c-container"
      onKeyDown={[Function]}
    >
      <div
        className=" css-yk16xz-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className=" css-1hwfws3"
        >
          <div
            className=" css-1wa3eu0-placeholder"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chart-line fa-w-16 "
              data-icon="chart-line"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M496 384H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM464 96H345.94c-21.38 0-32.09 25.85-16.97 40.97l32.4 32.4L288 242.75l-73.37-73.37c-12.5-12.5-32.76-12.5-45.25 0l-68.69 68.69c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0L192 237.25l73.37 73.37c12.5 12.5 32.76 12.5 45.25 0l96-96 32.4 32.4c15.12 15.12 40.97 4.41 40.97-16.97V112c.01-8.84-7.15-16-15.99-16z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
          <div
            className="css-w8afj7-Input"
          >
            <div
              className=""
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                disabled={false}
                id="react-select-8-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "label": "input",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className=" css-1g48xl4-IndicatorsContainer"
        >
          <span
            className=" css-1okebmr-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            className=" css-tlfecz-indicatorContainer"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-tj5bde-Svg"
              focusable="false"
              height={20}
              viewBox="0 0 20 20"
              width={20}
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-gzVnrw fgiriP"
  >
    <p>
      0
    </p>
    <div
      className=" css-2b097c-container"
      onKeyDown={[Function]}
    >
      <div
        className=" css-yk16xz-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className=" css-1hwfws3"
        >
          <div
            className=" css-1uccc91-singleValue"
          >
            American
          </div>
          <div
            className="css-w8afj7-Input"
          >
            <div
              className=""
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                disabled={false}
                id="react-select-9-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "label": "input",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className=" css-1g48xl4-IndicatorsContainer"
        >
          <span
            className=" css-1okebmr-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            className=" css-tlfecz-indicatorContainer"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-tj5bde-Svg"
              focusable="false"
              height={20}
              viewBox="0 0 20 20"
              width={20}
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-ifAKCX fFHKWo"
  >
    <button
      className="sc-bwzfXH nTFNk"
      onClick={[Function]}
    >
      Clear
    </button>
    <button
      className="sc-bwzfXH nTFNk"
      disabled={true}
      onClick={[Function]}
    >
      Send To Betslip
    </button>
  </div>,
]
`;
